cypress-io/cypress
# chore: add schema infrastructure and switch CDP client to be passed into protocol directly

### Additional details


This PR along with https://github.com/cypress-io/cypress-services/pull/5486:

* Sets up the infrastructure to add schema declarations and consumptions via `kysely` and `zod`
* Passes in the cdp client to the cloud code rather than relying on the cloud code to create a new connection. This will enable reuse of all of the retries, reconnects, and cleanup that are done with the connection
* Passes in the better-sqlite3 db instead of just a path to file

### Steps to test


- Clone the `cypress-services` repo and checkout `ryanm/spike/schema`
  - Run `yarn`
  - Run `yarn watch` in `packages/app-capture-protocol`
- Set `CYPRESS_LOCAL_PROTOCOL_PATH` to the path to `cypress-services/packages/app-capture-protocol/dist/index.js`
- Clone the `cypress` repo  and checkout `ryanm/spike/schema`
  - Run `yarn`
  - Execute `cypress run` on a project in record mode with `DEBUG` set to `cypress:*protocol*` and ensure you can see the lifecycle events

You can test the schema typing by making modifications to these queries:

* https://github.com/cypress-io/cypress-services/pull/5486/files#diff-6fea80cc14e191751ca6f34233cc36a73ed34fe14795bdf1d449f64202ca46fbR28-R46
* https://github.com/cypress-io/cypress-services/pull/5486/files#diff-6fea80cc14e191751ca6f34233cc36a73ed34fe14795bdf1d449f64202ca46fbR48-R55

### How has the user experience changed?


n/a

### PR Tasks


- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
